compositor_render/src/wgpu/ctx.rs

```diff
@@ -69,6 +69,23 @@ impl WgpuCtx {
             error!("Selected adapter is CPU based. Aborting.");
             return Err(CreateWgpuCtxError::NoAdapter);
         }
+        let required_features =
+            wgpu::Features::TEXTURE_BINDING_ARRAY | wgpu::Features::PUSH_CONSTANTS;
+        let optional_features =
+            wgpu::Features::SAMPLED_TEXTURE_AND_STORAGE_BUFFER_ARRAY_NON_UNIFORM_INDEXING
+                | wgpu::Features::UNIFORM_BUFFER_AND_STORAGE_TEXTURE_ARRAY_NON_UNIFORM_INDEXING;
+
+        let missing_critical_features = critical_features.difference(adapter.features());
+        if !missing_critical_features.is_empty() {
+            error!("Selected adapter or its driver does not support required wgpu features. Missing features: {missing_critical_features:?}).");
+            return Err(CreateWgpuCtxError::NoAdapter);
+        }
+        let missing_nominal_mode_features = nominal_mode_features.difference(adapter.features());
+        if !missing_nominal_mode_features.is_empty() {
+            error!("Selected adapter or its driver does not support optional wgpu features. Missing features: {missing_optional_features:?}).");
+        }
+        let required_features = critical_features
```

Comment:
There is no `critical_features` variable, this will no compile

Reply:
Solved in 6ecc7e356fd1d0eab9ee220c198dd067446f7103.
Note : with critical_features renamed to required_features as asked, I need to find a third name for the result of the computation of required + optional - missing_optional. I choose requested_features. I pass requested_features var to the required_feature arg of wgpu adapter.request_device() creating of slightly misleading naming thing. It seems reusing required_features name was worse.

Reply:
You did not have to change that variable name, rust supports shadowing and it's relatively common to use it, but requested_feature is fine too.